MetroscopeModelingLibrary/MultiFluid/HeatExchangers/CoolingTowerPoppewithSS.mo

```diff
@@ -0,0 +1,301 @@
+        extent={{-10,-10},{10,10}},
+        rotation=270,
+        origin={0,28})));
+  MetroscopeModelingLibrary.MoistAir.BoundaryConditions.Source
+                                                             air_outlet annotation (Placement(transformation(
+        extent={{-10,-10},{10,10}},
+        rotation=270,
+        origin={0,-32})));
+  MetroscopeModelingLibrary.MoistAir.BaseClasses.IsoPHFlowModel air_outlet_flow annotation (Placement(transformation(
+        extent={{-10,-10},{10,10}},
+        rotation=270,
+        origin={0,-64})));
+equation
+
+  // connectors
+  air_inlet_flow.P_out = Pin[1];
+  air_inlet_flow.Q = Q_cold_in;
+  air_inlet_flow.h = i_initial;
+  air_inlet.T_in = T_cold_in;
+  w_in = air_inlet.Xi_in[1];
+
+  air_outlet_flow.P_in = Pin[N_step];
+  air_outlet_flow.Q = Q_cold_out;
+  air_outlet_flow.h = i_final;
+  air_outlet.T_out = T_cold_out;
+  w_out = air_outlet.Xi_out[1];
+
+  water_inlet_flow.P_out = Pin[N_step];
+  water_inlet_flow.Q = Q_hot_in;
+  water_inlet_flow.T_in = T_hot_in;
+
+  water_outlet_flow.P_out = Pin[1];
+  water_outlet_flow.Q = Q_hot_out;
+  water_outlet_flow.T_in = T_hot_out;
+
+  // New Poppe Equations
+
+  deltaTw = (Tw[N_step] - Tw[1]) / (N_step - 1);
+
+  for n in 1:N_step loop
+    Tw[n] = T_hot_out + (T_hot_in-T_hot_out)*(n-1)/(N_step-1);
+    Ta[n] = MoistAir.T_phX(Pin[n], i[n], {w[n]});
+    w_sat[n] = MoistAir.xsaturation_pT(Pin[n], Ta[n]);
+  end for;
+
+  for n in 1:N_step-1 loop
+
+
+  if w[n] < w_sat[n] then
+     w[n+1] = w[n] + deltaTw * f(Tw[n], w[n], i[n], cp[n], Qw[n], Qa[n], Pin[n], Lef[n]);
+     i[n+1] = i[n] + deltaTw * g(Tw[n], w[n], i[n], cp[n], Qw[n], Qa[n], Pin[n], Lef[n]);
+     M[n+1]= M[n] + deltaTw * h(Tw[n+1], w[n+1], i[n+1], cp[n+1], Pin[n+1], Lef[n+1]);
+
+    Qw[n+1] = Qw[n] + Qa[n] * (w[n+1] - w[n]);
+
+    Qa[n+1] = Qa[n] * (1 + (w[n+1] - w[n]));
+
+    Lef[n+1] = Lef[n];
+
+    cp[n+1] = cp[n];
+
+    Pin[n+1] = Pin[n];
+
+   else
+     w[n+1] = w[n] + deltaTw * j(Tw[n], Ta[n], w[n], i[n], cp[n], Qw[n], Qa[n], Pin[n], Lef[n]);
+     i[n+1] = i[n] + deltaTw * k(Tw[n], Ta[n], w[n], i[n], cp[n], Qw[n], Qa[n], Pin[n], Lef[n]);
+     M[n+1]= M[n] + deltaTw * m(Tw[n+1], Ta[n+1], w[n+1], i[n+1], cp[n+1], Pin[n+1], Lef[n+1]);
+
+    Qw[n+1] = Qw[n] + Qa[n] * (w[n+1] - w[n]);
+
+    Qa[n+1] = Qa[n] * (1 + (w[n+1] - w[n]));
+
+    Lef[n+1] = Lef[n];
+
+    cp[n+1] = cp[n];
+
+    Pin[n+1] = Pin[n];
+
+  end if;
+
+  end for;
+
+  Me = hd * Afr / Qw[1];
+  M[N_step] = Me;
+  M[1] = 0;
+
+  w[1] = w_in;
+  w[N_step] = w_out;
+
+  i[1] = i_initial;
+  i[N_step] = i_final;
+
+  Qw[1] = Q_hot_out;
+  Qw[N_step] = Q_hot_in;
+  Qa[1] = Q_cold_in;
+  Qa[N_step] = Q_cold_out;
+
+  Lef[1] = 0.9077990913 * (((MoistAir.xsaturation_pT(Pin[1], T_cold_in)+0.622)/(w[1]+0.622))-1) / log((MoistAir.xsaturation_pT(Pin[1], T_cold_in)+0.622)/(w[1]+0.622));
+  cp[1] = WaterSteamMedium.specificHeatCapacityCp(water_inlet_flow.state_in);
+
+   // Drift Equation
```

Comment:
This equation is correct only for natural draft cooling tower. I think we should also add the equation that describe the fans behaviour with an if condition to selected the configuration. Something like:
```
if configuration = "natural draft" then
  ...
else
 ...
```
This has to be done also for the Merkel model.